safe_transaction_service/events/tests/test_queue_service.py

```diff
@@ -57,6 +57,17 @@ def test_send_unsent_messages(self):
             payload = f"not sent {i}"
             _, _, body = broker_connection.channel.basic_get(self.queue, auto_ack=True)
             self.assertEqual(json.loads(body), payload)
+        # Test unsent messages when pool limit is reached
```

Comment:
I think you should split the tests, by default no connection pool limit and then 2 tests with connection pool limit set